waiter/integration/waiter/deployment_errors_test.clj

```diff
@@ -15,9 +15,31 @@
             [waiter.client-tools :refer :all]
             [waiter.settings :as settings]))
 
+(deftest ^:parallel ^:integration-fast test-health-check-misconfigured
+  (testing-using-waiter-url
+    (let [headers {:x-waiter-name (rand-name)
+                   :x-waiter-health-check-url "/status-402"}
+          {:keys [headers body] :as response} (make-request-with-debug-info headers #(make-kitchen-request waiter-url %))
+          service-id (get headers "x-waiter-service-id")]
+      (is (not (nil? service-id)))
+      (assert-response-status response 503)
+      (is (str/starts-with? body (str "Deployment error: " (-> (waiter-settings waiter-url) :messages :health-check-misconfigured))))
+      (delete-service waiter-url service-id)))
+
+  (testing-using-waiter-url
```

Comment:
I don't think we need to repeat the `testing-using-waiter-url` call here, right?